up_client_socket/rust/utransport-socket/src/lib.rs

```diff
@@ -0,0 +1,420 @@
+ use up_rust::{UCode, UMessage, UMessageType, UStatus, UTransport, UUri};
+ 
+ use crate::constants::BYTES_MSG_LENGTH;
+ use crate::constants::DISPATCHER_ADDR;
+ use protobuf::Message;
+ use std::collections::hash_map::Entry;
+ use std::collections::HashSet;
+ use std::io::{Read, Write};
+ use std::net::TcpStream as TcpStreamSync;
+ use std::{
+     collections::HashMap,
+     sync::{Arc, Mutex},
+ };
+ use tokio::task;
+ use up_rust::ComparableListener;
+ 
+ pub struct UTransportSocket {
+     socket_sync: TcpStreamSync,
+     listener_map: Arc<Mutex<HashMap<UUri, HashSet<ComparableListener>>>>,
+ }
+ impl Clone for UTransportSocket {
+     fn clone(&self) -> Self {
+         UTransportSocket {
+             socket_sync: self
+                 .socket_sync
+                 .try_clone()
+                 .expect("issue in cloning sync socket"),
+             //  listener_map: self.listener_map.clone(),
+             listener_map: self.listener_map.clone(),
+         }
+     }
+ }
+ 
+ impl Default for UTransportSocket {
+     fn default() -> Self {
+         Self::new()
+     }
+ }
+ 
+ impl UTransportSocket {
+     #[must_use]
+     /// # Panics
+     ///
+     /// Will panic if issue connecting in sync socket
+     pub fn new() -> Self {
+         let _socket_sync: TcpStreamSync =
+             TcpStreamSync::connect(DISPATCHER_ADDR).expect("issue in connecting  sync socket");
+         let socket_sync = _socket_sync;
+         UTransportSocket {
+             socket_sync,
+             listener_map: Arc::new(Mutex::new(HashMap::new())),
+         }
+     }
+ 
+     /// # Panics
+     ///
+     /// Will panic if failed to parse message
+     pub fn socket_init(&mut self) {
+         loop {
+             // Receive data from the socket
+             let mut buffer: [u8; BYTES_MSG_LENGTH] = [0; BYTES_MSG_LENGTH];
+             let bytes_read = match self.socket_sync.read(&mut buffer) {
+                 Ok(bytes_read) => bytes_read,
+                 Err(e) => {
+                     dbg!("Socket error: {}", e);
+                     break;
+                 }
+             };
+ 
+             // Check if no data is received
+             if bytes_read == 0 {
+                 continue;
+             }
+ 
+             let umessage =
+                 UMessage::parse_from_bytes(&buffer[..bytes_read]).expect("Failed to parse message");
+ 
+             match umessage
+                 .attributes
+                 .type_
+                 .enum_value_or(UMessageType::UMESSAGE_TYPE_UNSPECIFIED)
+             {
+                 UMessageType::UMESSAGE_TYPE_PUBLISH => {
+                     dbg!("calling handle publish....");
+                     let _ = self.check_on_receive(&umessage.attributes.source, &umessage);
+                 }
+                 UMessageType::UMESSAGE_TYPE_NOTIFICATION => todo!(),
+                 UMessageType::UMESSAGE_TYPE_UNSPECIFIED | UMessageType::UMESSAGE_TYPE_RESPONSE => (),
```

Comment:
Will Response messages also be handled in this PR before merging?

Reply:
that will come later